query-connector/src/app/(pages)/userManagement/components/DataProvider.tsx

```diff
@@ -0,0 +1,154 @@
+"use client";
+
+import { createContext, useState } from "react";
+
+type SubjectType = "Members" | "Query";
+
+interface UserManagementData {
+  TeamQueryEditSection: {
+    isOpen: boolean;
+    title: string;
+    subtitle: string;
+    placeholder: string;
+    subjectData: unknown;
+    subjectType: SubjectType;
+  };
+}
+
+interface UserManagementContext extends UserManagementData {
+  OpenEditSection: (
+    title: string,
+    subtitle: string,
+    subjectType: SubjectType,
+    subjectId: string,
+  ) => void;
+  CloseEditSection: () => void;
+  HandleSearch: (searchFilter: string) => void;
+  HandleMemberUpdate: () => void;
+  HandleQueryUpdate: () => void;
+}
+
+const initData: UserManagementData = {
+  TeamQueryEditSection: {
+    isOpen: false,
+    title: "",
+    subtitle: "",
+    placeholder: "Search",
+    subjectData: [],
+    subjectType: "Members",
+  },
+};
+
+export const DataContext = createContext<UserManagementContext>({
```

Comment:
nit: could we rename these `Data*` vars to something more descriptive - (ie `UserContext`?) In a world where we have multiple contexts for multiple things, having more explicit names might make it more clear which context we're using.

Reply:
I have made this change to add the prefix `UserManagement` to all components and objects related to handle the state of the page. I do see that the naming gets verbose. Maybe we should revisit the prefix pattern because one could argue that the context for the different components can be inferred but its location (in this case the UserManagement folder) in the same way that there are multiple page.tsx and layout.tsx files. If multiple components/objects need to be imported with the same name we could use aliases.  

Anyway I added the prefix so will marking this as resolved!